### PR TITLE
(#221) showing applicant timeline

### PR DESCRIPTION
**Pull request by rathorevaibhav**
#221 

Changes include:

1. Getting applicant timeline and with all `applications` and `applicationRound` details. Sorting the timeline based on the order of date for each event.
2. Resume icon and reason for eligibility was not showing up in the application edit screen. Fixed.

### CODE REVIEW

**Comment by bohraji** [comment]
Checked this PR :

- [ ] For duplication.
- [ ] For performance. 
- [ ] Usage of recursive methods, loops and algorithms.
- [ ] For readability and understanding. 
- [ ] For functions with multiple responsibilities.
- [ ] For scope of refactoring. 
- [ ] For coupling
- [ ] For alternative approaches to the implementation. 
- [ ] For how the change impacts other parts of the system.
- [ ] For validations. client side and server side validations are present.

File: resources/views/hr/application/timeline.blade.php [hunk 1]
```diff
@@ -1,36 +1,32 @@
...
+                            <span data-toggle="modal" data-target="#autoresponder_mail" class="modal-toggler-text text-primary">Auto-respond mail from system</span>
+                            @include('hr.application.auto-respond', ['applicant' => $application->applicant, 'application' => $application])
+                        @endif
+                        @break
+                    @case('round-conducted')
+                        @php
+                            $applicationRound = $item['applicationRound'];
+                        @endphp
+                        <b><u>{{ date(config('constants.display_date_format'), strtotime($applicationRound->conducted_date)) }}</u></b><br>
+                        {{ $applicationRound->round->name }} for {{ $applicationRound->application->job->title }} conducted by {{ $applicationRound->conductedPerson->name }}<br>
```
Comment (bohraji) [inline resources/views/hr/application/timeline.blade.php]: Please eager load all the relationship that are being used.

File: app/Models/HR/Applicant.php [hunk 2]
```diff
@@ -55,4 +55,35 @@ public function applications()
...
+    /**
+     * Get the timeline for an applicant
+     * @return array
+     */
+    public function timeline()
+    {
+        $this->load('applications', 'applications.applicationRounds');
+        $timeline = [];
+        foreach ($this->applications as $application) {
+            $timeline[] = [
```
[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply app/Models/HR/Applicant.php]: Agreed.

**Comment by rathorevaibhav** [comment]
@bohraji all feedback implemented. Assigning back fo review.

**Pull request merged**